query/src/main/scala/filodb/query/exec/MetadataExecPlan.scala

```diff
@@ -164,7 +164,56 @@ final case class TopkCardPresenter(k: Int) extends RangeVectorTransformer {
 
 final case class LabelValuesDistConcatExec(queryContext: QueryContext,
                                            dispatcher: PlanDispatcher,
-                                           children: Seq[ExecPlan]) extends MetadataDistConcatExec
+                                           children: Seq[ExecPlan]) extends MetadataDistConcatExec {
+  /**
+   * Compose the sub-query/leaf results here.
+   */
+  override final def compose(childResponses: Observable[(QueryResponse, Int)],
+                        firstSchema: Task[ResultSchema],
+                        querySession: QuerySession): Observable[RangeVector] = {
+    qLogger.debug(s"NonLeafMetadataExecPlan: Concatenating results")
+    val taskOfResults = childResponses.map {
+      case (QueryResult(_, schema, result, _, _, _), _) => (schema, result)
+      case (QueryError(_, _, ex), _)         => throw ex
+    }.toListL.map { resp =>
+      val colType = resp.head._1.columns.head.colType
+      if (colType == MapColumn) {
+        val metadataResult = scala.collection.mutable.Set.empty[Map[ZeroCopyUTF8String, ZeroCopyUTF8String]]
+        resp.foreach { result =>
+          val rv = result._2
+          metadataResult ++= rv.head.rows.map { rowReader =>
+            val binaryRowReader = rowReader.asInstanceOf[BinaryRecordRowReader]
+            rv.head match {
+              case srv: SerializedRangeVector =>
+                srv.schema.toStringPairs (binaryRowReader.recordBase, binaryRowReader.recordOffset)
+                  .map (pair => pair._1.utf8 -> pair._2.utf8).toMap
+              case _ => throw new UnsupportedOperationException("Metadata query currently needs SRV results")
+            }
+          }
+        }
+        IteratorBackedRangeVector(new CustomRangeVectorKey(Map.empty),
+          UTF8MapIteratorRowReader(metadataResult.toIterator), None)
```

Comment:
Since this is a duplicate code, if possible move this to a common method and use the same in MetadataDistConcatExec as well.